R/ncdf.R

```diff
@@ -303,20 +311,21 @@ read_ncdf = function(.x, ..., var = NULL, ncsub = NULL, curvilinear = character(
   return(var)
 }
 
-.get_nc_projection <- function(atts, rep_var, coord_var) {
-  if(!is.null(atts)) {
-    nc_grid_mapping <- suppressWarnings(ncmeta::nc_grid_mapping_atts(atts, rep_var))
-  } else {
-    nc_grid_mapping <- list()
-  }
-
-  if(length(nc_grid_mapping) == 0) {
-    rep_coord_var <- coord_var[coord_var$variable == rep_var, ]
-
-    if(nrow(rep_coord_var) == 0) rep_coord_var[1, ] <- NA
+.get_grid_mapping <- function(atts, rep_var) {
+	if(!is.null(atts)) {
+		suppressWarnings(ncmeta::nc_grid_mapping_atts(atts, rep_var))
+	} else {
+		list()
+	}
+}
 
+.get_nc_projection <- function(atts, rep_var, rep_coord_var) {
 
-    if(.is_degrees(atts, rep_coord_var$X)) {
+  nc_grid_mapping <- .get_grid_mapping(atts, rep_var)
+  
+  if(length(nc_grid_mapping) == 0) {
+    
+    if(.is_degrees(atts, rep_coord_var)) {
```

Comment:
This warning is hit by the Euro Cordex text when a lon/lat curvilinear dimension is created but there is also a grid mapping to try and use.